src/prepare.ts

```diff
@@ -150,12 +154,16 @@ async function prepareElement(
         { ...memoElement, type: memoElement.type.type },
         errorHandler,
         context,
+        dispatcher,
       );
     }
     case ELEMENT_TYPE.FUNCTION_COMPONENT: {
       const functionElement = element as FunctionComponentElement<unknown>;
-      setDispatcherContext(context);
-      return [functionElement.type(functionElement.props), context];
+      setDispatcherContext(dispatcher, context);
+      registerDispatcher(dispatcher);
+      const children = functionElement.type(functionElement.props);
+      await Promise.all(popAwaitImmediatelyPromises(dispatcher));
```

Comment:
Well, yes, kind of. We actually can't await multiple `usePreparedEffect` with `awaitImmidiately = true` at all without the array, as they have to be awaited after the entire function body has run. So before adding the array it would only await the last one as each one would overwrite the previous.

I think the added complexity is worth it as it would create a very unexpected and hard to debug bug if we ever choose to have two effects with `awaitImmidiately = true`